### PR TITLE
[network] add microsoft.botservice to supported private endpoints operations

### PR DESCRIPTION
**Pull request by swagatmishra2007**
**Description**

We have implemented support for private endpoints feature on Microsoft.BotService RP. Since private endpoints protocol and api structure is dictated by the network team for all RP's, our implementation is the same as other RP's. Hence this PR just seeks to allow-list Microsoft.BotService RP to be invoked via the CLI for the standard network commands for private endpoints

**Testing Guide**

Tested the impacted commands on a private subscription, after setting up the CLI dev instructions - 
az network private-endpoint-connection show --id
az network private-endpoint-connection reject
az network private-endpoint-connection list
**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
network

File: src/azure-cli/azure/cli/command_modules/network/private_link_resource_and_endpoint_connections/custom.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 
 def register_providers():
+    _register_one_provider("Microsoft.BotService/botServices", "2021-03-01", True)
     _register_one_provider("Microsoft.Automation/automationAccounts", "2020-01-13-preview", True)
     _register_one_provider('Microsoft.AppConfiguration/configurationStores', '2020-06-01', True)
     _register_one_provider("Microsoft.Batch/batchAccounts", "2020-03-01", True)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/private_link_resource_and_endpoint_connections/custom.py:22]: Please put it in alphabetical order.


```suggestion
    _register_one_provider("Microsoft.Automation/automationAccounts", "2020-01-13-preview", True)
    _register_one_provider('Microsoft.AppConfiguration/configurationStores', '2020-06-01', True)
    _register_one_provider("Microsoft.Batch/batchAccounts", "2020-03-01", True)
    _register_one_provider("Microsoft.BotService/botServices", "2021-03-01", True)

```

**Comment by kairu-ms** [comment]
Please add tests in [this file](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py)

**Comment by swagatmishra2007** [comment]
> Please add tests in [this file](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/network/tests/latest/test_private_endpoint_commands.py)

@kairu-ms thanks for the suggest. I have added a test to go over all scenarios

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**